src/Raven.Server/Config/Categories/LogsConfiguration.cs

```diff
@@ -9,58 +10,78 @@ namespace Raven.Server.Config.Categories
         [DefaultValue(128)]
         [MinValue(16)]
         [SizeUnit(SizeUnit.Megabytes)]
-        [ConfigurationEntry("Logs.MaxFileSizeInMb", ConfigurationEntryScope.ServerWideOnly)]
-        public Size MaxFileSize { get; set; }
+        [ConfigurationEntry("Logs.ArchiveAboveSizeInMb", ConfigurationEntryScope.ServerWideOnly)]
+        public Size ArchiveAboveSize { get; set; }
 
-        [Description("The number of hours logs are kept before they are deleted")]
-        [DefaultValue(3 * 24)]
-        [MinValue(24)]
-        [TimeUnit(TimeUnit.Hours)]
-        [ConfigurationEntry("Logs.RetentionTimeInHrs", ConfigurationEntryScope.ServerWideOnly)]
-        public TimeSetting? RetentionTime { get; set; }
+        [DefaultValue(3)]
+        [ConfigurationEntry("Logs.MaxArchiveDays", ConfigurationEntryScope.ServerWideOnly)]
+        public int? MaxArchiveDays { get; set; }
 
-        [Description("The maximum log size after which older files will be deleted. No log files will be deleted if this configuration is not set.")]
         [DefaultValue(null)]
-        [MinValue(256)]
-        [SizeUnit(SizeUnit.Megabytes)]
-        [ConfigurationEntry("Logs.RetentionSizeInMb", ConfigurationEntryScope.ServerWideOnly)]
-        public Size? RetentionSize { get; set; }
+        [MinValue(0)]
+        [ConfigurationEntry("Logs.MaxArchiveFiles", ConfigurationEntryScope.ServerWideOnly)]
+        public int? MaxArchiveFiles { get; set; }
 
-        [Description("Determine whether to compress the log files")]
         [DefaultValue(false)]
-        [ConfigurationEntry("Logs.Compress", ConfigurationEntryScope.ServerWideOnly)]
-        public bool Compress { get; set; }
-
-        #region Microsoft Logs
-        [Description("Determine whether to disable Microsoft logs")]
-        [DefaultValue(true)]
-        [ConfigurationEntry("Logs.Microsoft.Disable", ConfigurationEntryScope.ServerWideOnly)]
-        public bool DisableMicrosoftLogs { get; set; }
-        
-        [Description("The path to the JSON configuration file for Microsoft logs")]
-        [ReadOnlyPath]
-        [DefaultValue("settings.logs.microsoft.json")]
-        [ConfigurationEntry("Logs.Microsoft.ConfigurationPath", ConfigurationEntryScope.ServerWideOnly)]
-        public PathSetting MicrosoftLogsConfigurationPath { get; set; }
-        #endregion
+        [ConfigurationEntry("Logs.EnableArchiveFileCompression", ConfigurationEntryScope.ServerWideOnly)]
+        public bool EnableArchiveFileCompression { get; set; }
+
+        [DefaultValue(LogLevel.Error)]
+        [ConfigurationEntry("Logs.Microsoft.MinLevel", ConfigurationEntryScope.ServerWideOnly)]
+        public LogLevel MicrosoftMinLevel { get; set; }
+
+        [DefaultValue(false)]
+        [ConfigurationEntry("Logs.ThrowExceptions", ConfigurationEntryScope.ServerWideOnly)]
+        public bool ThrowExceptions { get; set; }
+
+        [Description("Location of NuGet packages cache")]
+        [DefaultValue("Logs/NuGet")]
```

Comment:
Logs directory is probably not a good place to keep anything for a longer period of time

Reply:
Let's talk tomorrow about this